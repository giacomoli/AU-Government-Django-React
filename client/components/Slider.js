import React from 'react'

const SliderTrigger = ({ id, title, hasError }) => {
  const toggleSlider = () => {
    const slider = document.querySelector(`#${id}`)
    const caret = document.querySelector(`#${id}-container .arrow`)
    caret.classList.toggle('fa-caret-left')
    caret.classList.toggle('fa-caret-down')
    slider.classList.toggle('hidden-xs-up')
  }
  return (
    <button
      onClick={toggleSlider}
      type="button"
      className={`btn ${hasError ? 'btn-danger' : 'btn-success'} slider-trigger`}
    >
      {title} <i className="fa fa-caret-left arrow" aria-hidden="true" />
      <style jsx>{`
        .slider-trigger {
          width: 100%;
          text-align: left;
        }
      `}</style>
    </button>
  )
}

export default ({ id, children, title, initialState = 'closed', hasError = false }) => (
  <div>
    <div id={`${id}-container`} className="slider">
      <SliderTrigger id={id} title={title} hasError={hasError} />
      <div id={id} className={`slider-body p-3 ${initialState === 'closed' ? 'hidden-xs-up' : ''}`}>
        {children}
      </div>
    </div>
  </div>
)
