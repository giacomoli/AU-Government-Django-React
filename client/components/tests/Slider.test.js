import React from 'react'
import renderer from 'react-test-renderer'
import Slider from '../Slider'

/* eslint-disable */
describe('Slider', () => {

  it('should render Slider without data', () => {
    const wrapper = renderer.create(
        <Slider />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render Slider with data initialState(close)', () => {
    const wrapper = renderer.create(
        <Slider
          id="first"
          children={<h1>TEST</h1>}
          title="Test title"
        />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render Slider with data initialState(hidden)', () => {
    const wrapper = renderer.create(
        <Slider
          id="first"
          children={<h1>TEST</h1>}
          title="Test title"
          initialState="hidden-xs-up"
        />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
