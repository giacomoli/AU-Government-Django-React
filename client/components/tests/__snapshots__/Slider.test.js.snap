// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should render Slider with data initialState(close) 1`] = `
<div>
  <div
    className="slider"
    id="first-container"
  >
    <button
      className="jsx-1712863955 btn btn-success slider-trigger"
      onClick={[Function]}
      type="button"
    >
      Test title
       
      <i
        aria-hidden="true"
        className="jsx-1712863955 fa fa-caret-left arrow"
      />
    </button>
    <div
      className="slider-body p-3 hidden-xs-up"
      id="first"
    >
      <h1>
        TEST
      </h1>
    </div>
  </div>
</div>
`;

exports[`Slider should render Slider with data initialState(hidden) 1`] = `
<div>
  <div
    className="slider"
    id="first-container"
  >
    <button
      className="jsx-1712863955 btn btn-success slider-trigger"
      onClick={[Function]}
      type="button"
    >
      Test title
       
      <i
        aria-hidden="true"
        className="jsx-1712863955 fa fa-caret-left arrow"
      />
    </button>
    <div
      className="slider-body p-3 "
      id="first"
    >
      <h1>
        TEST
      </h1>
    </div>
  </div>
</div>
`;

exports[`Slider should render Slider without data 1`] = `
<div>
  <div
    className="slider"
    id="undefined-container"
  >
    <button
      className="jsx-1712863955 btn btn-success slider-trigger"
      onClick={[Function]}
      type="button"
    >
       
      <i
        aria-hidden="true"
        className="jsx-1712863955 fa fa-caret-left arrow"
      />
    </button>
    <div
      className="slider-body p-3 hidden-xs-up"
    />
  </div>
</div>
`;
